src/signing_key.rs

```diff
@@ -106,6 +111,59 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        assert_eq!(pki.algorithm.oid, "1.3.101.112".parse().unwrap());
+        assert_eq!(pki.algorithm.parameters, None);
+
+        SigningKey::try_from(pki.private_key)
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> PrivateKeyDocument {
+        let alg_info = AlgorithmIdentifier {
+            oid: OID,
+            parameters: None,
+        };
```

Comment:
Now I see you use this in multiple places. In that case, belay my previous suggestion and make a constant for the whole algorithm ID:

```rust
const ALGORITHM_ID: AlgorithmIdentifier = AlgorithmIdentifier {
    oid: OID,
    parameters: None,
};
```